official/nlp/bert/electraconfigs.py

```diff
@@ -0,0 +1,140 @@
+# Copyright 2019 The TensorFlow Authors. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# ==============================================================================
+"""The main BERT model and related functions."""
+
+from __future__ import absolute_import
+from __future__ import division
+from __future__ import print_function
+
+import copy
+import json
+import six
+import tensorflow as tf
+from official.nlp.bert import configs
+from official.modeling.hyperparams import base_config
+
+class ElectraConfig(base_config.Config):
```

Comment:
could you check the dataclass usages of Config?

Reply:
Does this mean I just have to add the "@dataclasses.dataclass" in front of ElectraConfig?

Reply:
I think we can either:
1. remove "@dataclasses.dataclass", and write ElectraConfig(object).
or
2. subclass Config in the following way:
https://github.com/tensorflow/models/blob/master/official/vision/image_classification/configs/base_configs.py#L69
The __init__ function will be auto-generated.